### DISCUSSION TITLE
Why doesn't the new documentation use AsyncValue.guard?

### DISCUSSION THREAD

**Discussion by higaski** [rrousselGit/riverpod#3634] [General]
I've just been browsing some parts of the new documentation and noticed that there is a new TODO list example using an AsyncNotifier in it. For some reason however this example does not use the static `AsyncValue.guard` method like the old example did. Is there any particular reason for this? 

Furthermore the old example is marked as outdated...? What exactly does this concern?

[New example](https://riverpod.dev/docs/essentials/side_effects) in the side effects documentation

```dart
  Future<void> addTodo(Todo todo) async {
    // The POST request will return a List<Todo> matching the new application state
    final response = await http.post(
      Uri.https('your_api.com', '/todos'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(todo.toJson()),
    );

    // We decode the API response and convert it to a List<Todo>
    List<Todo> newTodos = (jsonDecode(response.body) as List)
        .cast<Map<String, Object?>>()
        .map(Todo.fromJson)
        .toList();

    // We update the local cache to match the new state.
    // This will notify all listeners.
    state = AsyncData(newTodos);
  }
```

[Old example](https://riverpod.dev/docs/providers/notifier_provider) in the NotifierProvider documentation

```dart
  Future<List<Todo>> _fetchTodo() async {
    final json = await http.get('api/todos');
    final todos = jsonDecode(json) as List<Map<String, dynamic>>;
    return todos.map(Todo.fromJson).toList();
  }

  Future<void> addTodo(Todo todo) async {
    // Set the state to loading
    state = const AsyncValue.loading();
    // Add the new todo and reload the todo list from the remote repository
    state = await AsyncValue.guard(() async {
      await http.post('api/todos', todo.toJson());
      return _fetchTodo();
    });
  }
```

**Comment by rrousselGit** [comment]
Because in reality, when an "add todo" operation fails, you don't want the entire todo list view to behave as if it failed to obtain the list of todos.

You only want to show a snackbar or show the submit button in red. Not the whole page as error.

**Reply by higaski** [reply]
Thanks, makes sense.
